app/helpers/payment_details_helper.rb

```diff
@@ -19,33 +22,36 @@ def link_to_payment_details(text, options = {})
   end
 
   def payment_details_path(merchant_account = site_account, url_params = {})
-    return "" if ["bogus" ,""].include?(merchant_account.payment_gateway_type.to_s)
-    named_route = [:admin, :account, merchant_account.payment_gateway_type]
-    polymorphic_path(named_route, url_params)
+    payment_gateway_type = merchant_account.payment_gateway_type
+    return '' if [:bogus, nil].include?(payment_gateway_type)
+
+    polymorphic_path([:admin, :account, payment_gateway_type], url_params)
   end
 
   def edit_payment_details(merchant_account = site_account)
-    return "" if ["bogus" ,""].include?(merchant_account.payment_gateway_type.to_s)
-    polymorphic_url([:edit, :admin, :account, merchant_account.payment_gateway_type])
+    payment_gateway_type = merchant_account.payment_gateway_type
```

Comment:
Some further opportunities for refactoring:
- rename `edit_payment_details` to `edit_payment_details_path`

And by the way, searching for this name (to check if there is any name clash) I stumbled on this helper: https://github.com/3scale/porta/blob/master/lib/developer_portal/lib/developer_portal/controller_methods/payment_paths_methods.rb#L11-L23

Which also implements another thing I was going to suggest: extracting `[:bogus, nil].include?(payment_gateway_type)` to a separate method, to avoid duplication of logic.

Reply:
I don't think `#edit_payment_details` is used anymore, I will remove it altogether.

Reply:
https://github.com/3scale/porta/pull/3171